InvenTree/users/api.py

```diff
@@ -187,15 +186,32 @@ class GetAuthToken(APIView):
     def get(self, request, *args, **kwargs):
         """Return an API token if the user is authenticated
 
-        - If the user already has a token, return it
-        - Otherwise, create a new token
+        - If the user already has a matching token, delete it and create a new one
+        - Existing tokens are *never* exposed again via the API
+        - Once the token is provided, it can be used for auth until it expires
         """
+
         if request.user.is_authenticated:
-            # Get the user token (or create one if it does not exist)
-            token, created = Token.objects.get_or_create(user=request.user)
-            return Response({
+
+            user = request.user
+            name = request.query_params.get('name', '')
+
+            # Delete any matching tokens
+            ApiToken.objects.filter(user=user, name=name).delete()
```

Comment:
Comment: I think it would be prudent to log it a token was actually deleted for development (maybe log the identifier)

Reply:
Good call I'll add that.